Tools/TestWebKitAPI/Tests/IPC/IPCSerialization.mm

```diff
@@ -417,20 +424,43 @@ inline bool isEqual(WKDDActionContext *a, WKDDActionContext* b)
 }
 #endif
 
+#if USE(PASSKIT) && !PLATFORM(WATCHOS) && !PLATFORM(APPLETV)
+static BOOL wkSecureCoding_isEqual(id a, SEL, id b)
+{
+    RetainPtr<WKKeyedCoder> aCoder = adoptNS([WKKeyedCoder new]);
+    RetainPtr<WKKeyedCoder> bCoder = adoptNS([WKKeyedCoder new]);
+
+    [a encodeWithCoder:aCoder.get()];
+    [b encodeWithCoder:bCoder.get()];
+
+    return [[aCoder accumulatedDictionary] isEqual:[bCoder accumulatedDictionary]];
+}
+#endif // USE(PASSKIT) && !PLATFORM(WATCHOS) && !PLATFORM(APPLETV)
+
 inline bool operator==(const ObjCHolderForTesting& a, const ObjCHolderForTesting& b)
 {
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+        // These classes do not have isEqual: methods useful for our unit testing, so we'll swap it in ourselves.
+
+#if USE(PASSKIT) && !PLATFORM(WATCHOS) && !PLATFORM(APPLETV)
+        class_addMethod(PAL::getPKPaymentMethodClass(), @selector(isEqual:), (IMP)wkSecureCoding_isEqual, "v@:@");
```

Comment:
This looks awful.  Can't we just check the class and call a function like normal people?

Reply:
I would not propose all this for production code.
I'm very happy doing it in testing code. =D